validator/sawtooth_validator/journal/chain.py

```diff
@@ -505,6 +505,37 @@ def _verify_block(self, blkw):
         self._blocks_processing[blkw.block.header_signature] = validator
         self._executor.submit(validator.run)
 
+    def _remove_block_and_verify_descendants(self, parent_block):
+        """For a parent block, remove it from the pending block queue and
+        verify any immediate descendant blocks for it in the pending block
+        queue.
+
+        Args:
+            parent_block (BlockWrapper): The block that will be removed from
+                the pending block queue and immediate descendants will be
+                verified.
+
+        Returns:
+            The list of blocks that were submitted to block validator for
+            verification.  An empty list indicates that the block had
+            no immediate descendants in the pending block queue.
+        """
+
+        pending_blocks = \
+            self._blocks_pending.pop(parent_block.identifier, [])
+        if pending_blocks:
+            # One or more immediate descendant blocks are sitting in the
+            # pending block queue, we schedule them for verification.
+            LOGGER.debug(
+                'Verify descendant blocks: %s (%s)',
+                parent_block,
+                [block.identifier[:8] for block in pending_blocks])
+
+            for pending_block in pending_blocks:
+                self._verify_block(pending_block)
+
+        return pending_blocks
```

Comment:
Can do, although I had it return the list of pending blocks in case the caller cared about them.